### ISSUE TITLE
setUnion does not work on arrays of size 1 

### ISSUE THREAD

**Issue by ranjanan** [arrayfire/arrayfire#1660]
Unions of size 1 give the following error: 
```
ERROR: ArrayFire Error (202) : Invalid input argument
```

It works for arrays with 2 or more elements though. 

Ref https://github.com/JuliaComputing/ArrayFire.jl/issues/99

**Comment by pavanky** [comment]
Can you show some example code in C that fails ?

**Comment by 9prady9** [comment]
@pavanky @ranjanan Reported this on gitter earlier today and i found that the following code stub generates the error he was talking about.

```cpp
const int n = 1;
af::array A = af::randu(n, f32);
af::array B = af::randu(n, f32);
af::array C = af::setUnion(A, B, false);
```

If n = 1, earlier code generates the below error.

```
A
[1 1 1 1]
   Offset: 0
   Strides: [1 1 1 1]
    0.3990 
B
[1 1 1 1]
   Offset: 0
   Strides: [1 1 1 1]
    0.6720 
ArrayFire Exception (Invalid input argument:202):
In function af_err af_set_union(void**, af_array, af_array, bool)
In file src/api/c/set.cpp:82
Invalid argument at index 1
Expected: first_info.isVector()

In function af::array af::setUnion(const af::array&, const af::array&, bool)
In file src/api/cpp/set.cpp:38
terminate called after throwing an instance of 'af::exception'
  what():  ArrayFire Exception (Invalid input argument:202):
In function af_err af_set_union(void**, af_array, af_array, bool)
In file src/api/c/set.cpp:82
Invalid argument at index 1
Expected: first_info.isVector()
```

**Comment by pavanky** [comment]
@9prady9 can you fix the issue by adding `isScalar()` in `set.cpp` and then checking other functions that are using `isVector` too ? 

For example I see `convolve` is also using this. We may need to punt to scalar multiplication when the kernel is a scalar instead of launching a convolve kernel.

**Comment by 9prady9** [comment]
@pavanky I have checked for other files that are using `isVector` and all of the cases seems to be of legitimate usecases. I have fixed convolve checks.

**Comment by shehzan10** [comment]
Fixed in hotfix-3.4.2
